### DISCUSSION TITLE
Optimize CSS class names in production builds

### DISCUSSION THREAD

**Discussion by Timer** [vercel/next.js#62360] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

Next.js generates verbose CSS class names for development **and** production builds, thus increasing the production size unnecessarily.

## Describe the solution you'd like

Keep the verbose names in development for debugging, however, production should use single-character class names.

**Comment by samcx** [comment]
Moving this to <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> [discussions](https://github.com/vercel/next.js/discussions) since this is a feature request!

**Reply by andreasdc** [reply]
The feature looks super useful and well thought out.
Do we have an estimated timeline for when this could be implemented?
